### ISSUE TITLE
Build fails when exporting a type using MarkdownLayoutProps

### ISSUE THREAD

**Issue by phelipetls** [withastro/astro#5383]
### What version of `astro` are you using?

1.6.6

### Are you using an SSR adapter? If so, which one?

No

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

Exporting a type using `MarkdownLayoutProps` causes the following error.

```
[plugin:astro:build] Expected ">" but found "const"
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-yaiimj?file=src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by MoustaphaDev** [comment]
The correct usage of this it to *not export* the type

**Comment by phelipetls** [comment]
Got it. To clarify I exported it to stop the @typescript-eslint/no-unused-vars rule from complaining, what should I do instead? Should I just configure the rule to not apply for this type in astro files?

**Comment by MoustaphaDev** [comment]
Maybe you can use a `// @ts-ignore` ?

**Comment by phelipetls** [comment]
Yes, suppressing the error is what I'm doing now, with `// eslint-disable-next-line @typescript-eslint/no-unused-vars`, but I was wondering if there was a better way to handle this.

I opened the issue because the error message seemed to me like a weird one, thought it might be a bug, but feel free to close if this is expected. Thanks!

**Comment by MoustaphaDev** [comment]
Sorry I misread your message, I'm not aware of a better way to tackle this

**Comment by matthewp** [comment]
@Princesseuh can you take a look if this error message can be improved?

**Comment by Princesseuh** [comment]
This seems to be a compiler bug, maybe something to do with the export + the generic argument confusing the compiler?

**Comment by bholmesdev** [comment]
@Princesseuh Yeah that sounds right. I recall seeing this error message on generic types before 🤷
